### DISCUSSION TITLE
Running Tests

### DISCUSSION THREAD

**Discussion by Utsav-06** [destrex271/pgweb-testing-harness#30] [Q&A]
Hello @destrex271,

I have used **act -j run-tests** and this is the output.
![image](https://github.com/destrex271/pgweb-testing-harness/assets/111552902/d84874ca-f4aa-4d19-ac77-b558150c9f72)

It seems that the "run-tests" process has failed.
What am I doing wrong?

**Comment by destrex271** [comment]
act might have installed a different version of the ubuntu image. Mostly these errors can be resolved by removing sudo from the setup.sh script and maybe some small sections here and there. 
Maybe we can have a separate branch for devs to work on with the setup.sh script customized for act if the number of errors seems to be really large.

**Reply by Utsav-06** [reply]
Yeah it worked. But there was also an other issue like I have to change the sequence of setup file from LF to CRLF to be able to run on windows. 

Also, can you give me an idea about test running time? b'coz it's taking too much time on my machine.

**Reply by destrex271** [reply]
It should ideally finish in under 20 minutes but it can take longer depending on your internet connection. If you're internet connection is slow you can set up a new branch in your fork and run the workflows in GitHub actions only

Also if you just want to check one or two tests and see if they are running or if you just want to see how the entire process works you can change the pattern of files being picked up for testing in the setup.sh script.

Another point to remember is that you won't be able to save the reports as such when using act because iirc they don't support artifacts.

**Reply by Utsav-06** [reply]
That's the problem. I want the report file but it's taking decades to complete and on the other hand the artifacts are not showing any files in Actions.
